### ISSUE TITLE
Projections for SimulationUI

### ISSUE THREAD

**Issue by alex4200** [cnr-ibf-pa/hbp-bsp-issues#156]
Projections need to be implemented in the SimUI

**Comment by alex4200** [comment]
@wizmer : I suggest to talk to @romani79 about that...

**Comment by alex4200** [comment]
Lets all talk to Armando

**Comment by alex4200** [comment]
@mgeplf Any update on this issue?

**Comment by alex4200** [comment]
@mgeplf is getting feedback from @romani79

**Comment by alex4200** [comment]
From @mgeplf 

> waiting on GPFS/viz cluster being back and stable

**Comment by jdcourcol** [comment]
@romani79 : what is a proper timeline for that ?

**Comment by jdcourcol** [comment]
@mmigliore @romani79 : I know that we will add extra synapses to mimic the Schaffer collaterals. Should we add that into the simulation GUI, with these extra synapses ? or should we wait for proper projections ?

**Comment by jdcourcol** [comment]
By extra synapses, I mean BBPP42-262

**Comment by romani79** [comment]
Well, then they are projections. They are available for O1 and they will be available for the full-scale.
The only simplification we did is that all those synapses come from a single presynaptic cell. Anyway, the tools should not care about this detail.

**Comment by jdcourcol** [comment]
The tool has to activates these projections and displays this capability to the end user.
Then the question is how that should show up in the GUI. What is it that you want the end user to parametrize for these projections ?

**Comment by mmigliore** [comment]
As a first example, the user must be able to activate them in  
different ways (from fully synchronized to completely random and  
asynchronous).
Ideally, the user must be able to upload a file with an arbitrary  
pattern and weight for each synapse.

I think we need to have a brainstorming session on this.


jdcourcol <notifications@github.com> ha scritto:

**Comment by romani79** [comment]
I agree with Michele, but this protocol does not make sense with the current projections.
In fact, all the synapses (of the order of millions) have only 1 presynaptic cell. If we deliver a stimulus, all the synapses will be activated at the same time, which is useless.
At the moment, the only use case I am thinking is the possibility to switch on and off the spontaneous synaptic events of the projecting synapses. This feature should come together with the possibility to switch on and off the minis of the internal synapses.
Said that, more meaningful projections with the correct number of presynaptic cells will be available, and delivering patters of stimuli will be a nice feature to have in the GUI.

**Comment by jdcourcol** [comment]
So this is on hold as long as we don't have proper Schaffer collaterals.

**Comment by alex4200** [comment]
This issue might be done later in M18

**Comment by antonelepfl** [comment]
@romani79 Is there any update on this?

**Comment by alex4200** [comment]
Can be tested again with PizDaint

**Comment by alex4200** [comment]
Tested OK
